openBCIBoard.js

```diff
@@ -1,14 +1,486 @@
-var serialport = require('serialport');
-module.exports = {
-  streaming: false,
-  serial: null,
-  createBoard: function(board){
-    return new serialport.SerialPort(board.port, {
-      baudrate: board.baudrate,
-      parser: this.parseData
+'use strict';
+
+var k = require('./OpenBCIConstants');
+var OpenBCISample = require('./OpenBCISample');
+var serialPort = require('serialport');
+var EventEmitter = require('events').EventEmitter;
+var util = require('util');
+var stream = require('stream');
+
+
+
+
+function OpenBCIFactory() {
+    var factory = this;
+
+    var _options = {
+        baudrate: 115200,
+        daisy: false
+    };
+
+    /**
+     * Purpose: The initialization method to call first, before any other method.
+     * Author: AJ Keller (@pushtheworldllc)
+     */
+    function OpenBCIBoard(portName,options,connectImmediately) {
+        var self = this;
+        var args = Array.prototype.slice.call(arguments);
+
```

Comment:
Yea so the lines with spaces above and below them are setup type lines, such as dealing with the input to this method, the lines with no spacing are initialization lines, such as just setting a badPackets to 0.